### PR TITLE
feat: remove duplicate emails ignoring case in Personalization

### PR DESCRIPTION
**Pull request by DougCal**
Fixes https://github.com/sendgrid/sendgrid-python/issues/788

This removes duplicate emails on setting the Personalization object, ignoring case.

### CODE REVIEW

**Comment by DougCal** [comment]
This currently doesn't work for lists of strings or lists of tuples.  Wasn't sure if they're needed for this feature, but let me know what you think of them and the expected behavior for if they're used.  Currently, this will raise an `AttributionError`.

**Review by eshanholtz — CHANGES_REQUESTED** [review]
Generally this looks good, but doesn't work when using just the `add_to`/`add_cc`/`add_bcc` functions:
```python
p = Personalization()
to_email = To('test+to@example.com', 'Example To Name 0')
p.add_to(to_email)
p.add_to(to_email)
```

Since there are several ways of setting/adding emails to these fields, and each set would result in a call to `_get_unique_recipients`, why not just use that function once in each of the getters? For example:
```python
@property
def tos(self):
   return _get_unique_recipients(self._tos)
```

**Comment by DougCal** [comment]
Going to get into updating this soon.  Been extremely busy this last month, but going to make time for this when I can.  Haven't forgotten this!

**Comment by DougCal** [comment]
@eshanholtz ready for re-review!  Loved the suggestion- cleaned up the code a lot.  I tested making requests locally to be sure it prevented the duplicate email error and it worked quite well.

File: test/test_mail_helpers.py [hunk 1]
```diff
@@ -379,6 +379,224 @@ def test_error_is_not_raised_on_to_emails_includes_bcc_cc(self):
...
+        self.maxDiff = None
+
+        p = Personalization()
+        cc_email = Cc('test+to0@example.com', 'Example To Name 0')
+        p.add_cc(cc_email)
+        p.add_cc(cc_email)
+
+        expected = [cc_email.get()]
+
+        self.assertEqual(expected, p.ccs)
```
Comment (thinkingserious) [inline test/test_mail_helpers.py:598]: These look to be duplicates of:

- https://github.com/sendgrid/sendgrid-python/pull/924/files#diff-dfb342158840d4c80c26b95c9be2c3fa7bba624850d288105c5462ce991014daR382
- https://github.com/sendgrid/sendgrid-python/pull/924/files#diff-dfb342158840d4c80c26b95c9be2c3fa7bba624850d288105c5462ce991014daR403
- https://github.com/sendgrid/sendgrid-python/pull/924/files#diff-dfb342158840d4c80c26b95c9be2c3fa7bba624850d288105c5462ce991014daR424

Comment (DougCal) [reply test/test_mail_helpers.py:598]: Semantically yes, but they are testing a different helper method.  Would you prefer to have them to make sure the helper methods that `add_email` uses works properly (`add_to`, `add_bcc`, and `add_cc`), or just have unit tests for `add_email` and not `add_to` and etc.?

Comment (eshanholtz) [reply test/test_mail_helpers.py:865]: I agree we should test them separately to ensure all of them work as expected, since it's all unique code.

**Review by eshanholtz — APPROVED** [review]
LGTM! Thank you for the contribution!

**Pull request merged**